### DISCUSSION TITLE
Tensor cannot be modified after creation

### DISCUSSION THREAD

**Discussion by scusimple** [deepjavalibrary/djl#1500] [Q&A]
Hi guys, I am new to DJL. I've Googled but no results. I want to replace the new value with NDArray.set function, like this:

>> val manager = NDManager.newBaseManager()
>> var array = manager.arange(1,10)
>> array.set(NDIndex("0"), 2)

but i get a Exception: Exception in thread "main" java.lang.UnsupportedOperationException: Tensor cannot be modified after creation
I‘m working for Windows 11 with Intellij IDEA
gradle file has added dependencies like this:
>> implementation "ai.djl:api:0.6.0"
>> implementation "ai.djl.tensorflow:tensorflow-native-auto:2.4.1"
>> runtimeOnly "ai.djl.tensorflow:tensorflow-engine:0.15.0"

I also check the TfNDArrayIndexer.set function source code, like this:
`public void set(NDArray array, NDIndexFullSlice fullSlice, Number value) {
    throw new UnsupportedOperationException("Tensor cannot be modified after creation");
}`

It's there any way to be able to modify NDArray with given value after creation?
Am I missing something trivial here?  Thanks for any help!

**Comment by frankfliu** [comment]
@scusimple 

Can you try PyTorch engine or MXNet engine?

For TensorFlow, we didn't implement modify the tensor by index. see: https://github.com/deepjavalibrary/djl/blob/master/engines/tensorflow/tensorflow-engine/src/main/java/ai/djl/tensorflow/engine/TfNDArrayIndexer.java#L64

It's not straightforward to add this feature for TensorFlow, but I think it should be possible for CPU and non-string type of tensor.

**Reply by scusimple** [reply]
thanks a lot, I will try MXNet instend of it
